File: HelloWorld/task3/JavaDfsAndBfs/graphs/src/main/java/dfs_and_bfs1/graphs/Graph.java
package dfs_and_bfs1.graphs;
import java.util.*;

public class Graph<Type> {
	public Graph(int size)
	{
		vertexes = new ArrayList<Type>(size);
		edges = new ArrayList<Vector<Integer>>(size);
		for (int i = 0; i < size; ++i)
		{
			vertexes.add(null);
			edges.add(new Vector<Integer>());
		}
	}
	
	public<AnotherType> Graph(Graph<AnotherType> graph)
	{
		vertexes = new ArrayList<Type>(graph.vertexes.size());
		edges = new ArrayList<Vector<Integer>>(graph.edges);
		for (int i = 0; i < graph.vertexes.size(); ++i)
		{
			vertexes.add(null);
		}
	}
	
	public Type getVertex(int number)
	{
		return vertexes.get(number);
	}
	
	public void changeVertex(int number, Type newValue)
	{
		vertexes.set(number, newValue);
	}
	
	public void addEdge(int fromVertex, int toVertex)
	{
		edges.get(fromVertex).add(toVertex);
		edges.get(toVertex).add(fromVertex);
	}
	
	public Vector<Integer> getVertexNeighbours(int number)
	{
		return edges.get(number);
	}
	
	public int size()
	{
		return vertexes.size();
	}
	
	private ArrayList<Type> vertexes;
	private ArrayList<Vector<Integer>> edges; 
}

File: HelloWorld/task3/CsDfsAndBfs/CsDfsAndBfs/Entities/Graph.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CsDfsAndBfs.Entities
{
    public class Graph<Type>
    {
		public Graph(int size)
		{
			vertexes = new List<Type>(size);
			edges = new List<List<int>>(size);
			for (int i = 0; i < size; ++i)
			{
				vertexes.Add(default);
				edges.Add(new List<int>(size));
			}
		}

		public void ConstructByGraph<AnotherType>(Graph<AnotherType> graph)
		{
			vertexes = new List<Type>(graph.vertexes.Count);
			edges = new List<List<int>>(graph.edges);
			for (int i = 0; i < graph.vertexes.Count; ++i)
			{
				vertexes.Add(default);
			}
		}

		public Type GetVertex(int number)
		{
			return vertexes[number];
		}

		public void ChangeVertex(int number, Type newValue)
		{
			vertexes[number] = newValue;
		}

		public void AddEdge(int fromVertex, int toVertex)
		{
			edges[fromVertex].Add(toVertex);
			edges[toVertex].Add(fromVertex);
		}

		public List<int> GetVertexNeighbours(int number)
		{
			return edges[number];
		}

		public int Size()
		{
			return vertexes.Count;
		}

		private List<Type> vertexes;
        private List<List<int>> edges;
    }
}
